include/GafferScene/MeshTessellate.h

```diff
@@ -0,0 +1,86 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2024, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "GafferScene/ObjectProcessor.h"
+
+#include "Gaffer/StringPlug.h"
+#include "Gaffer/TypedPlug.h"
+
+namespace GafferScene
+{
+
+class GAFFERSCENE_API MeshTessellate : public ObjectProcessor
+{
+
+	public :
+
+		MeshTessellate( const std::string &name=defaultName<MeshTessellate>() );
+		~MeshTessellate() override;
+
+		Gaffer::IntPlug *divisionsPlug();
+		const Gaffer::IntPlug *divisionsPlug() const;
+
+		Gaffer::BoolPlug *calculateNormalsPlug();
+		const Gaffer::BoolPlug *calculateNormalsPlug() const;
+
+		Gaffer::BoolPlug *tessellatePolygonsPlug();
+		const Gaffer::BoolPlug *tessellatePolygonsPlug() const;
+
+		Gaffer::IntPlug *schemePlug();
+		const Gaffer::IntPlug *schemePlug() const;
```

Comment:
Yeah, I found it kind of awkward as well. Having `scheme` trump `tessellatePolygons` does make sense, and I'm fine with renaming it to `schemeOverride`, or perhaps `overrideScheme`?

The naming of `tessellatePolygons` is trickier. `useBilinearForPolygons` doesn't fair very well in a future where we expose meshScheme more directly to users, in which case it would be `useBilinearForBilinear`? I wonder if we could more easily describe the inverse ... something like `skipPolygons`, default True? Maybe that doesn't help much ... I'm expecting this to be used rarely, and am not too concerned about it ... if `useBilinearForPolygons` is what you want, I'm OK with that.

Reply:
Thinking about this more, I think what I don't like about `useBilinearForPolygons` is that either way, `bilinear` is the scheme used for polygons, unless overridden. Polygons are identified by their interpolation being set to `linear`, and if we don't tessellate them, their shape matches a linear interpolation. The actual intent of this parameter is "skipTessellatingThingsIfTheirShapeWontChange(AndTheSchemeIsntOverridden)" or "forceTessellatingThingsEvenIfTheirShapeWontChange", but I can't think of a catchy name for that. I do kind of feel like "tessellatePolygons" is the most reasonable name brought up so far.

Reply:
> useBilinearForPolygons doesn't fair very well in a future where we expose meshScheme more directly to users, in which case it would be useBilinearForBilinear?

I think technically it would need to be called `useBilinearForNone`. Our `interpolation==linear` has always meant "just polys, no need to subdivide", which is equivalent to USD's `scheme==none`. Whereas USD's `scheme==bilinear` means "I am a subdiv surface, just with bilinear subdivision rules" (I think), and doesn't really have an equivalent in Cortex.

> Having scheme trump tessellatePolygons does make sense, and I'm fine with renaming it to schemeOverride, or perhaps overrideScheme?

OK, we're agreed on that at least, and we can grey out `tessellatePolygons` (or whatever it gets called) when `scheme` is non-default. `schemeOverride` sounds OK to me, but to me `overrideScheme` sounds like a bool rather than a string. I suppose we should also consider what we'd call any future override for faceVaryingLinear or whatever - maybe "override"  makes that too wordy?